rust-runtime/smithy-types/src/instant/format.rs

```diff
@@ -463,52 +458,24 @@ pub mod iso_8601 {
     fn format_nanos(into: &mut String, nanos: u32) {
         if nanos > 0 {
```

Comment:
can you add a precondition (or docs / both) that `nanos < 1_000_000_000`? If nanos is greater than that, this hard crashes with a divide by zero. It's safe right now since it's only called with `.subsec_nanos()` but just want to be future proof in a world where someone refactors this.